### ISSUE TITLE
Exception with no parameter name is found

### ISSUE THREAD

**Issue by xaviv** [Http-Multipart-Data-Parser/Http-Multipart-Data-Parser#58]
This form raises an exception because there is no parameter "name":
--BOUNDARY
Content-Type: application/octet-stream
Content-Disposition: attachment; filename=202_1708_09066660000001000000010016_20170527005018.data
Content-Length: 1549

Obviously the form is malformed, but the parser should be resillient to that error.

Stacktrace:
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at HttpMultipartParser.StreamingMultipartFormDataParser.ParseFilePart(Dictionary`2 parameters, RebufferableBinaryReader reader)
   at HttpMultipartParser.StreamingMultipartFormDataParser.ParseSection(RebufferableBinaryReader reader)
   at HttpMultipartParser.StreamingMultipartFormDataParser.Parse(RebufferableBinaryReader reader)
   at HttpMultipartParser.StreamingMultipartFormDataParser.Run()
   at HttpMultipartParser.MultipartFormDataParser..ctor(Stream stream, String boundary, Encoding encoding, Int32 binaryBufferSize)
   at HttpMultipartParser.MultipartFormDataParser..ctor(Stream stream)

**Comment by anton-makarevich** [comment]
I've created a small PR with a workaround for this issue 
https://github.com/Vodurden/Http-Multipart-Data-Parser/pull/61

**Comment by Jericho** [comment]
I know this issue is quite old but a related issue was recently resolved which impacts the issue you reported. 

The problem which was resolved is that the name of the file was used as an identifier which caused problems if multiple files had the same name  (#81) or multiple files had a blank name (#59). This issue was resolved by no longer relying on the name as an identifier which means that the `name` parameter should no longer be a required parameter.

I will make the necessary change so that `name` is not required and publish this change in next release.

Again, I know you reported this issue a long time ago (almost 3 years!) and appreciate your patience. I recently took over stewardship of this project and I'm trying to address issues  outstanding issues.
